tests/src/test/scala/com/nvidia/spark/rapids/OrcScanSuite.scala

```diff
@@ -33,4 +34,66 @@ class OrcScanSuite extends SparkQueryCompareTestSuite {
 
   testSparkResultsAreEqual("Test ORC count chunked by bytes", fileSplitsOrc,
     new SparkConf().set(RapidsConf.MAX_READER_BATCH_SIZE_BYTES.key, "100"))(frameCount)
+
+  testSparkResultsAreEqual("schema-can-prune dis-order read schema",
+    frameFromOrcWithSchema("schema-can-prune.orc", StructType(Seq(
+      StructField("c2_string", StringType),
+      StructField("c3_long", LongType),
+      StructField("c1_int", IntegerType))))) { frame => frame }
+
+  testSparkResultsAreEqual("schema-can-prune dis-order read schema 1",
+    frameFromOrcWithSchema("schema-can-prune.orc", StructType(Seq(
+      StructField("c2_string", StringType),
+      StructField("c1_int", IntegerType),
+      StructField("c3_long", LongType))))) { frame => frame }
+
+  testSparkResultsAreEqual("schema-can-prune dis-order read schema 2",
+    frameFromOrcWithSchema("schema-can-prune.orc", StructType(Seq(
+      StructField("c3_long", LongType),
+      StructField("c2_string", StringType),
+      StructField("c1_int", IntegerType))))) { frame => frame }
+
+  testSparkResultsAreEqual("schema-can-prune dis-order read schema 3",
+    frameFromOrcWithSchema("schema-can-prune.orc", StructType(Seq(
+      StructField("c3_long", LongType),
+      StructField("c2_string", StringType))))) { frame => frame }
+
+  testSparkResultsAreEqual("schema-can-prune dis-order read schema 4",
+    frameFromOrcWithSchema("schema-can-prune.orc", StructType(Seq(
+      StructField("c2_string", StringType),
+      StructField("c1_int", IntegerType))))) { frame => frame }
+
+  testSparkResultsAreEqual("schema-can-prune dis-order read schema 5",
+    frameFromOrcWithSchema("schema-can-prune.orc", StructType(Seq(
+      StructField("c3_long", LongType),
+      StructField("c1_int", IntegerType))))) { frame => frame }
+
+  /**
+   * We can't compare the results from CPU and GPU, since CPU will get in-correct result
+   * see https://github.com/NVIDIA/spark-rapids/issues/3060
```

Comment:
IIUC the "can't be pruned" case is when the column names appear to be the same as the case when ORC doesn't provide column names (which was true for old versions of ORC in the early Hive days).  In those cases, I don't think Spark can fix this because there aren't any column names to match up if the specified schema doesn't match the file order.

Reply:
Yes, you're correct. So if it's not a Spark bug, Should I merge this PR? @jlowe

Reply:
Per your test case, even without this fix, the CPU and GPU still produce different outputs. So personally I prefer to merge it.

Reply:
Agreed, this seems to not break anything and improves behavior in a corner case.